openeo/extra/stac_job_db.py

```diff
@@ -0,0 +1,303 @@
+import concurrent
+import datetime
+import logging
+from typing import Iterable, List, Union
+
+import geopandas as gpd
+import numpy as np
+import pandas as pd
+import pystac
+import requests
+from pystac import Collection, Item
+from pystac_client import Client
```

Comment:
as discussed, pystac_client is an optional dependency at the moment.

you should document that at https://github.com/Open-EO/openeo-python-client/blob/47172cdb479756f0a5c1d9f53f0005b1a9675929/docs/installation.rst?plain=1#L82-L94